include/misc/poset.h

```diff
@@ -0,0 +1,95 @@
+// VolEsti (volume computation and sampling library)
+
+// Copyright (c) 2012-2020 Vissarion Fisikopoulos
+// Copyright (c) 2018-2020 Apostolos Chalkis
+// Copyright (c) 2020-2021 Vaibhav Thakkar
+
+// Contributed and/or modified by Vaibhav Thakkar, as part of Google Summer of Code 2021 program.
+
+// Licensed under GNU LGPL.3, see LICENCE file
+
+#ifndef POSET_H
+#define POSET_H
+
+#include <iostream>
+#include <vector>
+
+
+class Poset {
+public:
+    typedef std::pair<unsigned int, unsigned int> RT; // relation type
+    typedef std::vector<RT> RV;  // relations vector
+
+private:
+    unsigned int n;  // elements will be from 0 to n-1
+    RV order_relations;     // pairs of form a <= b
+
+public:
+    Poset(unsigned int _n, RV& _order_relations) : 
+        n(_n), order_relations(verify(_order_relations, n))
+    {
+    }
+
+
+    // verify if the relations are valid
+    static RV verify(const RV& relations, unsigned int n)
```

Comment:
currently the verify function does a very naive check of whether the relations have elements in [0, n-1]. I think adding a test for it will make sense when an addition check of cyclic relations are checked, which can be done in a future PR.